Add cmf_find_translation twig function

EmmanuelVella opened the pull request:
Hello there,

This PR just adds a `cmf_find_translation` twig function. Usage example :

```
{% set document = cmf_find_translation('/cms/foo', locale) %}
```

dbu commented:
looks good. using this as in your example would be less than ideal design because you mix code (the hardcoded path in the template) with editable content. what if an admin moves the document?

can you please rebase on 1.3 ? that should get us rid of the styleci warning. 
and can you please do a PR on symfony-cmf-docs to add the function to the list of twig functions?

dbu commented on Templating/Helper/CmfHelper.php:
```diff
@@ -168,6 +168,19 @@ public function find($path)
+     */
+    public function findTranslation($path, $locale)
+    {
+        return $this->getDm()->findTranslation(null, $path, $locale);
```
just wondering: it could be convenient if this also accepts a document in addition to the path. so that i could get the translation of a document i already have, without worrying about how to get the path from the document.

you would start with is_string($path) and if not see if its a managed document and use the dm to read the path of the document... but that can also be added later when somebody actually needs it.

dbu commented:
btw the documentation should mention that the document will stay in the last requested language for the rest of the request. and that there is language fallback, and one might want to check the language of the document that was found to see if the requested language existed.

EmmanuelVella commented:
Hello David,

Thank you for your feedback ! I just have : 
- rebased my PR
- added some phpdoc
- added support for path and document in `findTranslation` method
- add `fallback` parameter to `findTranslation``method

What do you think ? I will add documentation as soon as it's ok for you.

dbu commented:
looks good, i think the fallback flag makes a lot of sense in this context. there is a slight overlap with cmf_document_locales (you could check if your locale is in the locales instead of disabling fallback) but i think twig functions are about convenience so its ok. and its not an exact overlap like if we had a cmf_translation_exists or something. (not saying we would need that, cmf_find_translation is good enough, you can check the result for null without real overhead in the code)

EmmanuelVella commented:
Thank you !

dbu commented:
tagged 1.3.1 with this feature. will update master from 1.3 now.
